packages/tools/devtools/devtools-browser-extension/src/popup/PopupScript.ts

```diff
@@ -28,4 +51,24 @@ browser.tabs.query({ active: true, currentWindow: true }, (tab) => {
 		'To use the Fluid Devtools, open the browser Devtools pane (F12) and click the "Fluid Developer Tools" tab.';
 
 	document.body.append(popupElement);
+	initializePopupView(popupElement, tabId).then(() => {
+		console.debug(`Rendered popup for tab ${tabId}!`);
+	}, console.error);
 });
+
+/**
+ * Renders the Fluid Popup view into the provided target element.
+ *
+ * @param target - The element into which the popup view will be rendered.
+ */
+export async function initializePopupView(target: HTMLElement, tabId: number): Promise<void> {
+	const backgroundServiceConnection = await BackgroundConnection.Initialize({
+		// TODO: devtools-panel-specific source
+		messageSource: extensionMessageSource,
+		tabId,
+	});
+
+	ReactDOM.render(React.createElement(PopupView, { backgroundServiceConnection }), target, () => {
+		console.log("Rendered Popup view!");
```

Comment:
```suggestion
		console.debug("Rendered Popup view!");
```

Reply:
Actually, we can probably just remove the callback parameter here. The caller in `PopupScript.ts` already does console logging when the promise is resolved, so we're effectively double-logging for render completion currently.